### ISSUE TITLE
How to take screenshot only once of the live stream in the beginning?

### ISSUE THREAD

**Issue by mohsalsaleem** [arut/nginx-rtmp-module#1230]
I found this snippet in another issue.

```
 recorder preview {
        record keyframes;
        record_max_frames 4;
        record_path /var/rec;
        record_interval 30s;

        exec_record_done ffmpeg -i $path -vcodec png -vframes 1 -an -f rawvideo -s 320x240 -ss 00:00:01 -y /var/preview/$name.png;
    }
```
But, this produces the screenshot only after the stream has ended. How do I capture a picture at the beginning of the stream or after set interval only once and be done with it?

**Comment by karlisk** [comment]
You could try the `on_publish` directive.
